src/lsdb/nested/datasets/generation.py

```diff
@@ -1,9 +1,16 @@
+import numpy as np
+from astropy.coordinates import Angle, SkyCoord
+from cdshealpix.nested import healpix_to_lonlat
 from nested_pandas import datasets
 
+import lsdb
+from lsdb.core.search import BoxSearch, ConeSearch, PixelSearch
 from lsdb.nested.core import NestedFrame
 
 
-def generate_data(n_base, n_layer, npartitions=1, seed=None) -> NestedFrame:
+def generate_data(
+    n_base, n_layer, npartitions=1, seed=None, ra_range=(0.0, 360.0), dec_range=(-90, 90), search_filter=None
```

Comment:
Based on #736, can you go ahead and name this `region`?

Reply:
Done!